common/src/main/scala/com/gu/media/youtube/YouTubeAccess.scala

```diff
@@ -27,28 +28,27 @@ trait YouTubeAccess extends Settings with Logging {
   val disallowedVideos: Set[String] = getStringSet("youtube.videos.disallowed")
   val usePartnerApi: Boolean = getString("youtube.usePartnerApi").forall(_.toBoolean)
 
-  def clientId = getMandatoryString("youtube.clientId")
-  def clientSecret = getMandatoryString("youtube.clientSecret")
-  def refreshToken = getMandatoryString("youtube.refreshToken")
-
   def monetizationPolicyId = getMandatoryString("youtube.monetizationPolicyId")
   def trackingPolicyId = getMandatoryString("youtube.trackingPolicyId")
 
   def minDurationForAds: Long = getString("youtube.minDurationForAds").getOrElse("30").toLong
 
+  def serviceAccountCertPath = getMandatoryString("stage") match {
+    case "PROD" | "CODE" => "/etc/gu/youtube-service-account.json"
+    case _ => System.getProperty("user.home") + "/.gu/youtube-service-account.json"
+  }
+
   // Videos need to be at least 8 minutes to be eligible for midroll advertising
   // see https://support.google.com/youtube/answer/6175006?hl=en-GB
   val minDurationForMidroll: Long = 8 * 60L
 
   private val httpTransport = new NetHttpTransport()
   private val jacksonFactory = new JacksonFactory()
 
-  private val credentials: GoogleCredential = new GoogleCredential.Builder()
-    .setTransport(httpTransport)
-    .setJsonFactory(jacksonFactory)
-    .setClientSecrets(clientId, clientSecret)
-    .build
-    .setRefreshToken(refreshToken)
+
+  private lazy val credentials: GoogleCredential = GoogleCredential.fromStream(
+    new FileInputStream(serviceAccountCertPath)
+  ).createScoped(YouTubeScopes.all())
```

Comment:
Just to confirm, [these scopes](https://developers.google.com/resources/api-libraries/documentation/youtube/v3/java/latest/com/google/api/services/youtube/YouTubeScopes.html) matches what I used in `akash1810/google-oauth2-token-viewer`. I've this file on my machine:

```json
{
    "host": "http://localhost:8080",
    "callback": "http://localhost:8080/oauth2callback",
    "client_id": "ABC",
    "client_secret": "XYZ",
    "scopes": [
        "https://www.googleapis.com/auth/youtube",
        "https://www.googleapis.com/auth/youtube.force-ssl",
        "https://www.googleapis.com/auth/youtube.upload",

        "https://www.googleapis.com/auth/youtubepartner",
        "https://www.googleapis.com/auth/youtube.force-ssl",
        "https://www.googleapis.com/auth/youtubepartner-channel-audit"
    ]
}
```

Not for this PR, but I wonder if `YOUTUBEPARTNER_CHANNEL_AUDIT` can be dropped?

> View private information of your YouTube channel relevant during the audit process with a YouTube partner.

Reply:
Seems reasonable, but I agree, not for this PR